traffic_ops/testing/api/v2/deliveryservices_test.go

```diff
@@ -0,0 +1,602 @@
+	}
+	if ds.TRResponseHeaders == nil {
+		t.Errorf("expected TRResponseHeaders: %s, actual: nil", testDS.TRResponseHeaders)
+	} else if *ds.TRResponseHeaders != testDS.TRResponseHeaders {
+		t.Errorf("expected TRResponseHeaders: %s, actual: %s", testDS.TRResponseHeaders, *ds.TRResponseHeaders)
+	}
+	if ds.ConsistentHashRegex == nil {
+		t.Errorf("expected ConsistentHashRegex: %s, actual: nil", testDS.ConsistentHashRegex)
+	} else if *ds.ConsistentHashRegex != testDS.ConsistentHashRegex {
+		t.Errorf("expected ConsistentHashRegex: %s, actual: %s", testDS.ConsistentHashRegex, *ds.ConsistentHashRegex)
+	}
+	if ds.ConsistentHashQueryParams == nil {
+		t.Errorf("expected ConsistentHashQueryParams: %v, actual: nil", testDS.ConsistentHashQueryParams)
+	} else if !reflect.DeepEqual(ds.ConsistentHashQueryParams, testDS.ConsistentHashQueryParams) {
+		t.Errorf("expected ConsistentHashQueryParams: %v, actual: %v", testDS.ConsistentHashQueryParams, ds.ConsistentHashQueryParams)
+	}
+	if ds.MaxOriginConnections == nil {
+		t.Errorf("expected MaxOriginConnections: %d, actual: nil", testDS.MaxOriginConnections)
+	} else if *ds.MaxOriginConnections != testDS.MaxOriginConnections {
+		t.Errorf("expected MaxOriginConnections: %d, actual: %d", testDS.MaxOriginConnections, *ds.MaxOriginConnections)
+	}
+
+	// GET 1.1, verify 1.3 and 1.4 fields are nil
+	data := tc.DeliveryServicesNullableResponse{}
+	if err = makeV11Request(http.MethodGet, "deliveryservices/"+strconv.Itoa(*ds.ID), nil, &data); err != nil {
+		t.Errorf("cannot GET 1.1 deliveryservice: %s", err.Error())
+	}
+	respDS := data.Response[0]
+	if !dsV13FieldsAreNil(respDS) || !dsV14FieldsAreNil(respDS) {
+		t.Error("expected 1.3 and 1.4 values to be nil, actual: non-nil")
+	}
+
+	// GET 1.3, verify 1.3 fields are non-nil and 1.4 fields are nil
+	data = tc.DeliveryServicesNullableResponse{}
+	if err = makeV13Request(http.MethodGet, "deliveryservices/"+strconv.Itoa(*ds.ID), nil, &data); err != nil {
+		t.Errorf("cannot GET 1.3 deliveryservice: %s", err.Error())
+	}
+	respDS = data.Response[0]
+	if dsV13FieldsAreNil(respDS) {
+		t.Error("expected 1.3 values to be non-nil, actual: nil")
+	}
+	if !dsV14FieldsAreNil(respDS) {
+		t.Error("expected 1.4 values to be nil, actual: non-nil")
+	}
+	if _, err = TOSession.DeleteDeliveryService(strconv.Itoa(*ds.ID)); err != nil {
+		t.Errorf("cannot DELETE deliveryservice: %s", err.Error())
+	}
+
+	ds.ID = nil
+	dsBody, err := json.Marshal(ds)
+	if err != nil {
+		t.Errorf("cannot POST deliveryservice, failed to marshal JSON: %s", err.Error())
+	}
+	dsV11Body, err := json.Marshal(ds.DeliveryServiceNullableV11)
+	if err != nil {
+		t.Errorf("cannot POST deliveryservice, failed to marshal JSON: %s", err.Error())
+	}
+
+	// POST 1.3 w/ 1.4 data, verify 1.4 fields were ignored
+	postDSResp := tc.CreateDeliveryServiceNullableResponse{}
+	if err = makeV13Request(http.MethodPost, "deliveryservices", bytes.NewBuffer(dsBody), &postDSResp); err != nil {
```

Comment:
This whole bit about creating a v1.1 Delivery Service and POST-ing it to the 1.3 API then checking that the resulting object properly ignored newer fields shouldn't be included, because this can't/won't test the v1.1/1.3 API